### DISCUSSION TITLE
Parameter passing from "query results" query to underlying query

### DISCUSSION THREAD

**Discussion by K-Lohse** [getredash/redash#6618] [Q&A] [answered]
Hi everyone,

Is there a possibility to pass a parameter of a "query results" query down to a "subquery"

Usecase:
I got some limited JSON APIs that I query and combine with query results. The JSON apis have a setting that I want to parameterize. 
For example I have two JSON Apis with a branch parameter:
JSON query_1: `http://myserver.com/{{ branch }}/getMisraIssues`
JSON query_2: `http://myserver.com/{{ branch }}/getFileOwners`
Query 1 gives me a list of issues in the given branch with the filename where the issues occours,
Query 2 will give me a list of filenames with their owner.
I need the combination of both to see the number of issues per owner:
Query results query: 
`select query_2.ComponetOwner, count(*) as NrOfIssues from query_1 
inner join query_2 on query_1.Filename = query_2.Filename
group by query2.ComponentOwner`

Unfortunately I did not find an option to pass the parameter {{branch}} from the "query results" query to the underlying JSON queries. I did find an old discussion about this:
https://discuss.redash.io/t/thoughts-on-adding-support-for-queries-with-parameters-in-query-results-data-source/1709
But it looks like this was not implemented?

What I hoped for, was some choice like when directly embedding a parametrized query_1 in a dashboard, or at least a way to recreate such a choice.

I assume the workaround is to make a set of api-requests for the different parameters, union them and then make a parametrized query on the unioned data.

**Comment by guidopetri** [comment, marked as answer]
That's correct - we don't support queries with parameters in QRDS yet. I don't recall the exact details but it seemed possible to implement, just nobody got around to it, so if you'd like to open a PR, feel free :)

**Comment by anthonycorletti** [comment]
This is supported now correct? https://redash.io/help/user-guide/querying/query-results-data-source/

**Reply by ypJuanPN** [reply]
Did you manage to make it work?
